modules/Bio/EnsEMBL/Variation/DBSQL/PublicationAdaptor.pm

```diff
@@ -206,22 +206,72 @@ sub fetch_all_by_dbID_list {
 sub fetch_all_by_Variation {
     my $self    = shift;
     my $var_obj = shift;
-    
+
     if(!defined($var_obj) || ref($var_obj ne 'Variation')) {
-        throw("variation argument is required");
+        throw("Variation argument is required");
     }
-    
+
+    my $variation_id = $var_obj->dbID;
+
     my @pub;
     my $publication_id;
+    my $data_source_attrib;
+    my $attrib_id_to_value = {};
 
-    my $sth = $self->prepare(qq{SELECT  publication_id from variation_citation where variation_id = ?  });
-    $sth->execute($var_obj->dbID);
-    $sth->bind_columns(\$publication_id);
+    my $sth = $self->prepare(qq{SELECT  publication_id, data_source_attrib from variation_citation where variation_id = ?  });
+    $sth->execute($variation_id);
+    $sth->bind_columns(\$publication_id, \$data_source_attrib);
     while ($sth->fetch()){
-        push @pub, $self->fetch_by_dbID($publication_id)
+
+      my $publication = $self->fetch_by_dbID($publication_id);
+
+      if($data_source_attrib) {
+
+        if($data_source_attrib =~ /,/) {
+          my @sources = split /,/, $data_source_attrib;
```

Comment:
a split also works if the string you are planning to split doesn't contain the split character. You don't need to have a separate case for when $data_source_attrib doesn't contain a ','